components/search/src/types.rs

```diff
@@ -0,0 +1,197 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/// The list of possible application names that are currently supported.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchApplicationName {
+    Firefox = 1,
+    FirefoxAndroid = 2,
+    FocusAndroid = 3,
+    FirefoxIOS = 4,
+    FocusIOS = 5,
+}
+
+impl SearchApplicationName {
+    pub fn as_str(&self) -> &'static str {
+        match self {
+            SearchApplicationName::Firefox => "firefox",
+            SearchApplicationName::FirefoxAndroid => "firefox-android",
+            SearchApplicationName::FocusAndroid => "focus-android",
+            SearchApplicationName::FirefoxIOS => "firefox-ios",
+            SearchApplicationName::FocusIOS => "focus-ios",
+        }
+    }
+}
+
+/// The list of possible update channels for a user's build.
+/// Use `default` for a self-build or an unknown channel.
+#[derive(Debug, uniffi::Enum)]
+pub enum SearchUpdateChannel {
```

Comment:
I noticed this is named `SearchUpdateChannel`, and I was wondering about the reasoning behind that. To maintain consistency with the naming conventions in Desktop, do you think it might make sense to rename it to just channel?

Reply:
@linabutler tells me that Swift code doesn't support namespaces, so everything gets flattened into a single "module". Hence, if two different components have a type like `ApplicationName` or `Channel` then it'll refuse to build on Swift.

It doesn't always need to be prefixed with `Search` but it should be fairly guaranteed to be unique (e.g. I'm going to change `RefinedConfig` to `RefinedSearchConfig`).